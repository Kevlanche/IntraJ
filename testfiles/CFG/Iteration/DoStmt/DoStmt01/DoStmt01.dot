digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[18]"; "Opt[19]"; "Exit[20]"}
{ rank = same; "List[15]"; "Block[20]"}
{ rank = same; "List[21]"}
{ rank = same; "VarDeclStmt[22]"; "DoStmt[31]"; "VarDeclStmt[43]"}
{ rank = same; "Modifiers[23]"; "PrimitiveTypeAccess[25]"; "List[26]"; "Block[32]"; "VarAccess[42]"; "Modifiers[44]"; "PrimitiveTypeAccess[46]"; "List[47]"}
{ rank = same; "List[24]"; "VariableDeclarator[27]"; "List[33]"; "List[45]"; "VariableDeclarator[48]"}
{ rank = same; "List[28]"; "Opt[29]"; "VarDeclStmt[34]"; "List[49]"; "Opt[50]"}
{ rank = same; "BooleanLiteral[30]"; "Modifiers[35]"; "PrimitiveTypeAccess[37]"; "List[38]"}
{ rank = same; "List[36]"; "VariableDeclarator[39]"}
{ rank = same; "List[40]"; "Opt[41]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < DoStmt01 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[19]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[20]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[21]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[22]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[27]" [label="VariableDeclarator\n < boolean a >", fillcolor=white   style=filled  ]
"List[28]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[29]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[30]" [label="BooleanLiteral\n < false >", fillcolor=white   style=filled  ]
"DoStmt[31]" [label="DoStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[32]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[34]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[35]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[36]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[37]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[39]" [label="VariableDeclarator\n < int b >", fillcolor=white   style=filled  ]
"List[40]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[41]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[42]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"VarDeclStmt[43]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[44]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[45]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[46]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[47]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[48]" [label="VariableDeclarator\n < int c >", fillcolor=white   style=filled  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[50]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[20]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[18]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"VarDeclStmt[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"VarDeclStmt[22]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"VariableDeclarator[27]" -> "List[28]" [style=dashed, color=gray]
"Opt[29]" -> "BooleanLiteral[30]" [style=dashed, color=gray]
"VariableDeclarator[27]" -> "Opt[29]" [style=dashed, color=gray]
"List[26]" -> "VariableDeclarator[27]" [style=dashed, color=gray]
"VarDeclStmt[22]" -> "List[26]" [style=dashed, color=gray]
"List[21]" -> "VarDeclStmt[22]" [style=dashed, color=gray]
"Modifiers[35]" -> "List[36]" [style=dashed, color=gray]
"VarDeclStmt[34]" -> "Modifiers[35]" [style=dashed, color=gray]
"VarDeclStmt[34]" -> "PrimitiveTypeAccess[37]" [style=dashed, color=gray]
"VariableDeclarator[39]" -> "List[40]" [style=dashed, color=gray]
"VariableDeclarator[39]" -> "Opt[41]" [style=dashed, color=gray]
"List[38]" -> "VariableDeclarator[39]" [style=dashed, color=gray]
"VarDeclStmt[34]" -> "List[38]" [style=dashed, color=gray]
"List[33]" -> "VarDeclStmt[34]" [style=dashed, color=gray]
"Block[32]" -> "List[33]" [style=dashed, color=gray]
"DoStmt[31]" -> "Block[32]" [style=dashed, color=gray]
"DoStmt[31]" -> "VarAccess[42]" [style=dashed, color=gray]
"List[21]" -> "DoStmt[31]" [style=dashed, color=gray]
"Modifiers[44]" -> "List[45]" [style=dashed, color=gray]
"VarDeclStmt[43]" -> "Modifiers[44]" [style=dashed, color=gray]
"VarDeclStmt[43]" -> "PrimitiveTypeAccess[46]" [style=dashed, color=gray]
"VariableDeclarator[48]" -> "List[49]" [style=dashed, color=gray]
"VariableDeclarator[48]" -> "Opt[50]" [style=dashed, color=gray]
"List[47]" -> "VariableDeclarator[48]" [style=dashed, color=gray]
"VarDeclStmt[43]" -> "List[47]" [style=dashed, color=gray]
"List[21]" -> "VarDeclStmt[43]" [style=dashed, color=gray]
"Block[20]" -> "List[21]" [style=dashed, color=gray]
"Opt[19]" -> "Block[20]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[19]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[20]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[20]" -> "VariableDeclarator[48]" [color=red, constraint=false]
"VariableDeclarator[48]" -> "Exit[20]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[48]" -> "VarAccess[42]" [color=red, constraint=false]
"VarAccess[42]" -> "VariableDeclarator[39]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[42]" -> "VariableDeclarator[48]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[42]" -> "VariableDeclarator[39]" [color=red, constraint=false]
"VariableDeclarator[39]" -> "VarAccess[42]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[39]" -> "VariableDeclarator[27]" [color=red, constraint=false]
"VariableDeclarator[39]" -> "VarAccess[42]" [color=red, constraint=false]
"VariableDeclarator[27]" -> "VariableDeclarator[39]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[27]" -> "BooleanLiteral[30]" [color=red, constraint=false]
"BooleanLiteral[30]" -> "VariableDeclarator[27]" [color=blue, constraint=false, xlabel=" "] 
"BooleanLiteral[30]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "BooleanLiteral[30]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[18]" -> "Opt[19]" -> "Exit[20]" [style=invis]
"VarDeclStmt[22]" -> "DoStmt[31]" -> "VarDeclStmt[43]" [style=invis]
"Modifiers[23]" -> "PrimitiveTypeAccess[25]" -> "List[26]" [style=invis]
"List[28]" -> "Opt[29]" [style=invis]
"Block[32]" -> "VarAccess[42]" [style=invis]
"Modifiers[35]" -> "PrimitiveTypeAccess[37]" -> "List[38]" [style=invis]
"List[40]" -> "Opt[41]" [style=invis]
"Modifiers[44]" -> "PrimitiveTypeAccess[46]" -> "List[47]" [style=invis]
"List[49]" -> "Opt[50]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
52", fillcolor=1]
Edges [label="Total Nr. Edges
65", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
7", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
7", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}