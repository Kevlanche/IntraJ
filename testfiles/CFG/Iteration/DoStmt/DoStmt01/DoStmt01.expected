NodeName: Entry[13] Successors: [BooleanLiteral[30]]
NodeName: BooleanLiteral[30] Successors: [VariableDeclarator[27]]
NodeName: VariableDeclarator[27] Successors: [VariableDeclarator[39]]
NodeName: VariableDeclarator[39] Successors: [VarAccess[42]]
NodeName: VarAccess[42] Successors: [VariableDeclarator[39], VariableDeclarator[48]]
NodeName: VariableDeclarator[48] Successors: [Exit[20]]
NodeName: Exit[20] Successors: []

NodeName: Exit[20] Predecessor: [VariableDeclarator[48]]
NodeName: VariableDeclarator[48] Predecessor: [VarAccess[42]]
NodeName: VarAccess[42] Predecessor: [VariableDeclarator[39]]
NodeName: VariableDeclarator[39] Predecessor: [VariableDeclarator[27], VarAccess[42]]
NodeName: VariableDeclarator[27] Predecessor: [BooleanLiteral[30]]
NodeName: BooleanLiteral[30] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
